### ISSUE TITLE
About the DoH server

### ISSUE THREAD

**Issue by UjuiUjuMandan** [greatfire/envoy#86]
> Currently, DoH servers are hardcoded (dns.sb, quad9, cloudflare, and nextdns)

Reminder: In China the CCP blocked all the SNIs above (https://github.com/net4people/bbs/issues/68).

A possible circumvention is to use IP-only servers. For your information, below lists my personal use.

```
# DNS.SB
https://185.222.222.222/dns-query
https://45.11.45.11/dns-query

# Quad9
https://9.9.9.9/dns-query
https://149.112.112.112/dns-query

# OpenDNS
https://208.67.220.220/dns-query
https://208.67.222.222/dns-query
https://146.112.41.5/dns-query

# TWNIC
https://101.101.101.101/dns-query

# switch.ch
https://130.59.31.248/dns-query
https://130.59.31.251/dns-query

# Yandex
https://77.88.8.1/dns-query
https://77.88.8.8/dns-query

# AdGuard
https://94.140.14.140/dns-query
https://94.140.14.141/dns-query

# Cloudflare
https://162.159.36.1/dns-query
https://162.159.46.1/dns-query
https://1.1.1.1/dns-query
https://1.0.0.1/dns-query
https://162.159.132.53/dns-query
https://[2606:4700:4700::1111]/dns-query
https://[2606:4700:4700::1001]/dns-query
https://[2606:4700:4700::0064]/dns-query
https://[2606:4700:4700::6400]/dns-query

# TUNA
https://dns.tuna.tsinghua.edu.cn:8443/dns-query
```

**Comment by stevenmcdonald** [comment]
Hi, sorry we missed this until now. We have switched to IP based DNS (though I think we've only recently published the changes, oops)

Thanks for the list, that's more servers than we currently have

**Comment by UjuiUjuMandan** [comment]
@stevenmcdonald 

This is poste one year ago, now you might want to check the blocking targeting to these IP addresses.

* https://github.com/net4people/bbs/issues/295#issuecomment-2700908050

By a brief look, only the Russia Dad's Yandex is not blocked yet.

**Comment by stevenmcdonald** [comment]
@UjuiUjuMandan Yes, thanks, I tested out the servers briefly, I do want to run some more tests on the list though. I ended up adding switch.ch, yandex, and adguard to our list for now: https://github.com/greatfire/envoy/blob/master/android/envoy/src/main/kotlin/org/greatfire/envoy/EnvoyDns.kt#L15

It's been a few months since I tested, but blocking of DoH servers wasn't consistent. Sometimes/some regions, for example, some of the Clouldflare IPs will be blocked, sometimes not. Switch.ch, Yandex, and AdGuard were all working in my brief tests a couple weeks ago.

We're getting near a new release, so we'll be testing more. Anyway, thanks for the update, and sorry for the slow response, I'm bad at looking at github notifications
